src/planner.cc

```diff
@@ -659,7 +681,7 @@ void Planner::Impl::UpdateMotionProfile() {
 
   // Perform the backward pass. Compute and join the travel and deceleration
   // parts of the profile.
-  for (; buffer_index < planning_buffer_.size(); --buffer_index) {
+  for (; buffer_index >= last_planned_invariant_segment_; --buffer_index) {
```

Comment:
For semicolon-separated parts that are not needed in a `for`-loop, I usually add a `/**/`, just so that it is more visible.

```c++
for (/**/; buffer_index >= last_planned_invariant_segment_; --buffer_index) {
```

Not needed now, but could be considered in the future.